TESTS/mbed_hal/lp_us_tickers/main.cpp

```diff
@@ -13,98 +13,438 @@
+    timer.reset();
+    timer.start();
+    while (counter--) {
+        intf->set_interrupt(0);
+    }
+    timer.stop();
+
+    TEST_ASSERT(timer.read_us() < (NUM_OF_CALLS * (MAX_FUNC_EXEC_TIME_US + DELTA_FUNC_EXEC_TIME_US)));
+
+    /* ---- Test fire_interrupt function. ---- */
+    counter = NUM_OF_CALLS;
+    timer.reset();
+    timer.start();
+    while (counter--) {
+        intf->fire_interrupt();
+    }
+    timer.stop();
+
+    TEST_ASSERT(timer.read_us() < (NUM_OF_CALLS * (MAX_FUNC_EXEC_TIME_US + DELTA_FUNC_EXEC_TIME_US)));
+
+    /* ---- Test disable_interrupt function. ---- */
+    counter = NUM_OF_CALLS;
+    timer.reset();
+    timer.start();
+    while (counter--) {
+        intf->disable_interrupt();
+    }
+    timer.stop();
+
+    TEST_ASSERT(timer.read_us() < (NUM_OF_CALLS * (MAX_FUNC_EXEC_TIME_US + DELTA_FUNC_EXEC_TIME_US)));
+}
+
+
+utest::v1::status_t hf_ticker_setup(const Case *const source, const size_t index_of_case)
 {
     intf = get_us_ticker_data()->interface;
+
+    set_us_ticker_irq_handler(ticker_event_handler_stub);
+
     return greentea_case_setup_handler(source, index_of_case);
 }
 
 #if DEVICE_LOWPOWERTIMER
 utest::v1::status_t lp_ticker_setup(const Case *const source, const size_t index_of_case)
 {
     intf = get_lp_ticker_data()->interface;
+
+    set_lp_ticker_irq_handler(ticker_event_handler_stub);
+
     return greentea_case_setup_handler(source, index_of_case);
 }
 #endif
 
+utest::v1::status_t test_setup(const size_t number_of_cases)
+{
+    GREENTEA_SETUP(30, "default_auto");
+    return verbose_test_setup_handler(number_of_cases);
+}
+
 Case cases[] = {
-    Case("us ticker HAL - Init", us_ticker_setup, test_ticker_init),
-    Case("us ticker HAL - Read", us_ticker_setup, test_ticker_read),
-    Case("us ticker HAL - Read in the loop", us_ticker_setup, test_ticker_read_loop),
+      Case("hf ticker init is safe to call repeatedly", hf_ticker_setup, ticker_init_test),
+      Case("hf ticker frequency is non-zero and counter is at least 8 bits", hf_ticker_setup, ticker_info_test),
+      Case("hf ticker interrupt test", hf_ticker_setup, ticker_interrupt_test),
+      Case("hf ticker past interrupt test", hf_ticker_setup, ticker_past_test),
+      Case("hf ticker reschedule test", hf_ticker_setup, ticker_repeat_reschedule_test),
+      Case("hf ticker fire interrupt", hf_ticker_setup, ticker_fire_now_test),
+      Case("hf ticker overflow test", hf_ticker_setup, ticker_overflow_test),
+      Case("hf ticker increment test", hf_ticker_setup, ticker_increment_test),
+      Case("hf ticker speed test", hf_ticker_setup, ticker_speed_test),
 #if DEVICE_LOWPOWERTIMER
-    Case("lp ticker HAL - Init", lp_ticker_setup, test_ticker_init),
-    Case("lp ticker HAL - Read", lp_ticker_setup, test_ticker_read),
-    Case("lp ticker HAL - Read in the loop", lp_ticker_setup, test_ticker_read_loop),
+      Case("lp ticker init is safe to call repeatedly", lp_ticker_setup, ticker_init_test),
```

Comment:
Comment is on the test name: lp_us_tickers
Half of the test cases are not "lp_"...
I think it is nicer to have 1 us_tickers test for all targets, and 1 lp_us_tickers test for targets supporting LPT
It will be more understandable :-)
thx

Reply:
IMHO it makes sense to write 1 function by test for tests that fit us & lp ticker api and invoke them by changing the setup function.

Reply:
Current us/lp ticker test suit is consistent with the requirements provided in `mbed-os/hal/us_ticker_api.h` and  `mbed-os/hal/lp_ticker_api.h` header files. We have requirements which are common for us and lp tickers and requirements specific for each ticker, so test name `lp_us_tickers` means that we have here test cases common for both tickers. This is done to not duplicate test code. If the same function can be used to test us and lp ticker behaviour, then it is used in test case for lp and us ticker and setup function selects which ticker is tested.